lib/arjdbc/postgresql/adapter.rb

```diff
@@ -1012,10 +774,39 @@ module ActiveRecord::ConnectionAdapters
   remove_const(:PostgreSQLAdapter) if const_defined?(:PostgreSQLAdapter)
 
   class PostgreSQLAdapter < JdbcAdapter
+
+    # Store references to these methods so we can use them
+    # to overwrite the versions defined by
+    # ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements
+    # This makes it so we can use the bulk of the core logic while
+    # keeping the actual database access defined in this adapter
+    # This is hopefully temporary until this is cleaned up enough that we
+    # can extend ActiveRecord's AbstractAdapter or PostgreSQLAdapter instead of the JdbcAdapter
+    SAVED_METHODS = ['execute', 'exec_delete', 'exec_query', 'exec_update',
+                     'select_rows', 'select_value', 'select_values',
+                     'begin_db_transaction', 'commit_db_transaction', 'rollback_db_transaction',
+                     'begin_isolated_db_transaction', 'supports_transaction_isolation?',
+                     'create_savepoint', 'rollback_to_savepoint',
+                     'release_savepoint', 'current_savepoint_name']
+    SAVED_METHOD_PREFIX = 'jdbc_adapter'
+
+    SAVED_METHODS.each do |base_name|
+      alias_method :"#{SAVED_METHOD_PREFIX}_#{base_name}", base_name
+    end
+
+    # Keep this one separate because we don't want it to be automatically redefined
+    alias_method :"#{SAVED_METHOD_PREFIX}_exec_insert", :exec_insert
+
```

Comment:
@enebo The functionality for all of these is defined in the `JdbcAdapter` then it gets overwritten by the stuff from `DatabaseStatements` when it is included. Unfortunately, all of these methods have `PG` specific functionality tied to them, so we need to keep the functionality that was defined in `JdbcAdapter`. This was a quick and dirty way to do that without trying pull them all into a module and then have the same module included in the `JdbcAdapter` and again here in its subclass. I think in the future, we'll be able to use a similar setup to how the sqlite adapter works where it no longer subclasses `JdbcAdapter` and at that point it will make sense to move this stuff into a module or multiple modules, once we have a better idea of what all is going to need to be handled this way. I can try to bite that off now but thought it was better to keep this PR smaller and do that in a  future PR.